search-provider/eks-discovery-feed-provider.c

```diff
@@ -435,6 +469,166 @@ handle_article_card_descriptions (EksDiscoveryFeedDatabaseContentProvider *skele
 }
 
 static void
+get_word_of_the_day_content_cb (GObject *source,
+                                GAsyncResult *result,
+                                gpointer user_data)
+{
+  EkncEngine *engine = EKNC_ENGINE (source);
+  DiscoveryFeedQueryState *state = user_data;
+
+  g_application_release (g_application_get_default ());
+
+  GError *error = NULL;
+  g_autoptr(GSList) models = NULL;
+  GSList *shards = NULL;
+
+  if (!models_and_shards_for_result (engine,
+                                     state->provider->application_id,
+                                     result,
+                                     &models,
+                                     &shards,
+                                     &error))
+    {
+      g_dbus_method_invocation_take_error (state->invocation, error);
+      discovery_feed_query_state_free (state);
+      return;
+    }
+
+  GVariantBuilder builder;
+  g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{ss}"));
+
+  gint index = get_day_of_year () % g_slist_length (models);
+  EkncContentObjectModel *model = g_slist_nth (models, index)->data;
+
+  add_key_value_pair_from_model_to_variant (model, &builder, "title");
+  add_key_value_pair_from_model_to_variant (model, &builder, "synopsis");
+  add_key_value_pair_from_model_to_variant (model, &builder, "ekn-id");
+
+  g_dbus_method_invocation_return_value (state->invocation,
+                                         g_variant_new ("(a{ss})", &builder));
+  discovery_feed_query_state_free (state);
+}
+
+static gboolean
+handle_get_word_of_the_day (EksDiscoveryFeedDatabaseContentProvider *skeleton,
+                            GDBusMethodInvocation                   *invocation,
+                            gpointer                                 user_data)
+{
+    EksDiscoveryFeedDatabaseContentProvider *self = user_data;
+
+    if (!ensure_evergreen_app_proxy (self))
+      return TRUE;
+
+    EkncEngine *engine = eknc_engine_get_default ();
+
+    /* Build up tags_match_any */
+    GVariantBuilder tags_match_any_builder;
+    g_variant_builder_init (&tags_match_any_builder, G_VARIANT_TYPE ("as"));
+    g_variant_builder_add (&tags_match_any_builder, "s", "word_of_day");
+    GVariant *tags_match_any = g_variant_builder_end (&tags_match_any_builder);
+
+    /* Create query and run it */
+    g_autoptr(EkncQueryObject) query = g_object_new (EKNC_TYPE_QUERY_OBJECT,
+                                                     "tags-match-any", tags_match_any,
+                                                     "order", EKNC_QUERY_OBJECT_ORDER_DESCENDING,
+                                                     "limit", 365,
+                                                     "app-id", self->application_id,
+                                                     NULL);
```

Comment:
I discussed that with Nick the other day, we will not use content (WordOfTheDay) that is tight to a specific date. The logic here in place is only meant to not display the same quote or word again the next day. So that we have as much variation as possible.